shuup/admin/modules/media/views.py

```diff
@@ -210,36 +275,61 @@ def handle_upload(self):
         })
 
     def handle_post_rename_folder(self, data):
-        folder = Folder.objects.get(pk=data["id"])
+        shop = get_shop(self.request)
+        folder = _get_folder_query(shop).get(pk=data["id"])
+        if _is_folder_shared(folder):
+            message = _("Can not rename shared folder.")
+            return JsonResponse({"success": False, "message": message})
+
         folder.name = data["name"]
         folder.save(update_fields=("name",))
         return JsonResponse({"success": True, "message": _("Folder renamed.")})
 
     def handle_post_delete_folder(self, data):
-        folder = Folder.objects.get(pk=data["id"])
+        shop = get_shop(self.request)
+        folder = _get_folder_query(shop).get(pk=data["id"])
```

Comment:
handle `DoesNotExist` ?

Reply:
If these untranslated "not found" errors starts to happen then let's add the translation separately. These are catched at post already.